trin-core/src/portalnet/overlay_service.rs

```diff
@@ -365,7 +369,41 @@ impl OverlayService {
                     }
                 }
                 _ = OverlayService::bucket_maintenance_poll(self.protocol.clone(), &self.kbuckets) => {}
+                _ = bucket_refresh_interval.tick() => {
+                    debug!("Bucket refresh lookup");
+                    self.bucket_refresh_lookup();
+                }
+            }
+        }
+    }
+
+    /// Main bucket refresh lookup logic
+    fn bucket_refresh_lookup(&self) {
+        // Look at local routing table and select the largest 17 buckets.
+        // We only need the 17 bits furthest from our own node ID, because the closest 239 bits of
+        // buckets are going to be empty-ish.
+        let buckets = self.kbuckets.read();
+        let buckets = buckets.buckets_iter().enumerate().collect::<Vec<_>>();
+        let buckets = &buckets[239..];
```

Comment:
We probably want to move this to a constant somewhere, like `256-EXPECTED_NON_EMPTY_BUCKETS`, and a comment by there can explain how we got to 17 (estimating 100k nodes, which is more than 10x the ethereum mainnet node count). 2**17 is enough to put each node into a unique bucket by the 17th bucket index, with some hand-waving of even distribution assumptions.